### PR TITLE
networkmanager-openconnect: 1.2.8 -> 1.2.10

### PR DESCRIPTION
**Pull request by hcsch**
Changelog: https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/-/tags/1.2.10
https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/-/compare/1.2.8...1.2.10

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/networkmanager/openconnect/default.nix [hunk 1]
```diff
@@ -42,8 +42,9 @@ stdenv.mkDerivation rec {
     libxml2
     openconnect
     networkmanager
-  ] ++ lib.optionals withGnome [
     gtk3
+    webkitgtk # non-optional, requires gtk3
```
Comment (jtojnar) [inline pkgs/tools/networking/networkmanager/openconnect/default.nix:46]: Hmm, looks like it is needed for SSO: https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/-/commit/1e86b388f190e556aa9b934517b3a26cc64959d3

Also `webkitgtk_4_1` should probably be used.

Comment (hcsch) [reply pkgs/tools/networking/networkmanager/openconnect/default.nix]: Thanks for bringing that up, I didn't realise that 4.1 was also already packaged. Should've taken better care there, sorry.

**Comment by hcsch** [comment]
I've moved `gtk3` back into `buildInputs` only included `withGnome` since networkmanager-openconnect only needs it itself if configured that way, regardless of whether webkitgtk needs it.

**Review by jtojnar — APPROVED** [review]
Changelog and patch LGTM. Did not test.

**Comment by hcsch** [comment]
Re-tested with the changes I just pushed. I can't test SSO login, but using it for an AnyConnect VPN with a custom user agent works, GNOME settings displays stuff correctly as well. GNOME settings has one small issue though (which it has previously also had for me): The Apply button doesn't close the VPN specific settings, though it does save the changes. Somethings a bit broken there

**Comment by hcsch** [comment]
Oh, that was quick. Thanks a bunch :D

**Pull request merged**